### ISSUE TITLE
CuArrays 1.4.3 fails to initialize 

### ISSUE THREAD

**Issue by Sleort** [JuliaGPU/CuArrays.jl#482]
**Describe the bug**
CuArrays fails to initialize after updating from version 1.4.2 (which works fine) to 1.4.3. 

**To Reproduce**
```julia
julia> using CuArrays
[ Info: Recompiling stale cache file /home/troels/.julia/compiled/v1.2/CuArrays/7YFE0.ji for CuArrays [3a865a2d-5b23-5a0f-bc46-62713ec82fae]

julia> cu(rand(4))
ERROR: AssertionError: Cannot allocate before CuArrays has been initialized.
Stacktrace:
 [1] CuArray{Float32,1,P} where P(::UndefInitializer, ::Tuple{Int64}) at /home/troels/.julia/packages/CuArrays/BcKUl/src/memory.jl:120
 [2] convert at /home/troels/.julia/packages/CuArrays/BcKUl/src/array.jl:98 [inlined]
 [3] adapt_storage at /home/troels/.julia/packages/CuArrays/BcKUl/src/array.jl:240 [inlined]
 [4] adapt_structure at /home/troels/.julia/packages/Adapt/aeQPS/src/Adapt.jl:9 [inlined]
 [5] adapt at /home/troels/.julia/packages/Adapt/aeQPS/src/Adapt.jl:6 [inlined]
 [6] cu(::Array{Float64,1}) at /home/troels/.julia/packages/CuArrays/BcKUl/src/array.jl:288
 [7] top-level scope at REPL[2]:1
```

**Environment details (please complete this section)**
Details on Julia:
```
julia> versioninfo()
Julia Version 1.2.0
Commit c6da87ff4b (2019-08-20 00:03 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)
```

Julia packages:
 - CuArrays.jl: v1.4.3
 - CUDAnative.jl: v2.5.4

CUDA: toolkit and driver version
```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2019 NVIDIA Corporation
Built on Fri_Feb__8_19:08:17_PST_2019
Cuda compilation tools, release 10.1, V10.1.105
```

**Comment by maleadt** [comment]
https://github.com/JuliaGPU/CuArrays.jl/pull/483

**Comment by Sleort** [comment]
Fixed by v1.4.4
